### ISSUE TITLE
[shelly] RGBW2 : Updating colors from Coap doesn't work [solved]

### ISSUE THREAD

**Issue by markus7017** [openhab/openhab-addons#6949]
This exception is reported in the log:
```
2020-02-01 08:52:57.421 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526: CoIoT Message from /192.168.222.194:5683: NON-2.05   MID=39730, Token=null, OptionSet={"Uri-Path":["cit","s"], "Unknown (3332)":0x53485247425732233636323532362331, "Unknown (3412)":0x9600, "Unknown (3420)":0x0900}, "{"G":[[0,111, 255],[0,12".. 90 bytes
2020-02-01 08:52:57.422 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526: CoIoT Sensor data {"G":[[0,111, 255],[0,121, 255],[0,131, 0],[0,141, 255],[0,151, 63],[0,161, 0],[0,118,0]]}
2020-02-01 08:52:57.422 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526: 7†status updates received
2020-02-01 08:52:57.422 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526:  Sensor value[0]: Index=111, Value=255.0 (Red, Type=S, Range=0/255, Link=0: RGBW2)
2020-02-01 08:52:57.423 [TRACE] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526: Updating channel color.red from CoIoT, new value=255.0
2020-02-01 08:52:57.423 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526:  Sensor value[1]: Index=121, Value=255.0 (Green, Type=S, Range=0/255, Link=0: RGBW2)
2020-02-01 08:52:57.423 [TRACE] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526: Updating channel color.green from CoIoT, new value=255.0
2020-02-01 08:52:57.423 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526:  Sensor value[2]: Index=131, Value=0.0 (Blue, Type=S, Range=0/255, Link=0: RGBW2)
2020-02-01 08:52:57.424 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526:  Sensor value[3]: Index=141, Value=255.0 (White, Type=S, Range=0/255, Link=0: RGBW2)
2020-02-01 08:52:57.424 [TRACE] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526: Updating channel color.white from CoIoT, new value=255.0
2020-02-01 08:52:57.425 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526:  Sensor value[4]: Index=151, Value=63.0 (Gain, Type=S, Range=0/100, Link=0: RGBW2)
2020-02-01 08:52:57.425 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526:  Sensor value[5]: Index=161, Value=0.0 (VSwitch, Type=S, Range=0/1, Link=0: RGBW2)
2020-02-01 08:52:57.425 [TRACE] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526: Updating channel relay.output from CoIoT, new value=OFF
2020-02-01 08:52:57.425 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526:  Sensor value[6]: Index=118, Value=0.0 (Input, Type=S, Range=0/1, Link=0: RGBW2)
2020-02-01 08:52:57.426 [TRACE] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526:    map to input1 channel
2020-02-01 08:52:57.426 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526: Process 4 CoIoT channel updates
2020-02-01 08:52:57.426 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyrgbw2-662526:  Update[0] channel color#white, value=255.0
2020-02-01 08:52:57.427 [WARN ] [.californium.core.network.UdpMatcher] - error receiving request NON-CUSTOM_30 MID=39730, Token=, OptionSet={"Uri-Path":["cit","s"], "Unknown (3332)":0x53485247425732233636323532362331, "Unknown (3412)":0x9600, "Unknown (3420)":0x0900}, "{"G":[[0,111, 255],[0,12".. 90 bytes
java.lang.IllegalArgumentException: Value must be between 0 and 100
	at org.eclipse.smarthome.core.library.types.PercentType.validateValue(PercentType.java:57) ~[?:?]
	at org.eclipse.smarthome.core.library.types.PercentType.<init>(PercentType.java:52) ~[?:?]
	at org.eclipse.smarthome.core.library.types.DecimalType.as(DecimalType.java:152) ~[?:?]
	at org.eclipse.smarthome.core.internal.items.ItemStateConverterImpl.convertToAcceptedState(ItemStateConverterImpl.java:64) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.profiles.ProfileCallbackImpl.sendUpdate(ProfileCallbackImpl.java:134) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.profiles.SystemDefaultProfile.onStateUpdateFromHandler(SystemDefaultProfile.java:53) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.CommunicationManager.lambda$9(CommunicationManager.java:467) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.CommunicationManager.lambda$11(CommunicationManager.java:487) ~[?:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.CommunicationManager.handleCallFromHandler(CommunicationManager.java:483) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.CommunicationManager.stateUpdated(CommunicationManager.java:465) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.ThingManagerImpl$1.stateUpdated(ThingManagerImpl.java:168) ~[?:?]
	at org.eclipse.smarthome.core.thing.binding.BaseThingHandler.updateState(BaseThingHandler.java:245) ~[?:?]
	at org.eclipse.smarthome.core.thing.binding.BaseThingHandler.updateState(BaseThingHandler.java:264) ~[?:?]
	at org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.updateChannel(ShellyBaseHandler.java:732) ~[?:?]
	at org.openhab.binding.shelly.internal.coap.ShellyCoapHandler.handleStatusUpdate(ShellyCoapHandler.java:504) ~[?:?]
	at org.openhab.binding.shelly.internal.coap.ShellyCoapHandler.processResponse(ShellyCoapHandler.java:205) ~[?:?]
	at org.openhab.binding.shelly.internal.coap.ShellyCoapServer.lambda$0(ShellyCoapServer.java:122) ~[?:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:804) ~[?:?]
	at java.util.concurrent.CopyOnWriteArraySet.forEach(CopyOnWriteArraySet.java:425) ~[?:?]
	at org.openhab.binding.shelly.internal.coap.ShellyCoapServer.processResponse(ShellyCoapServer.java:122) ~[?:?]
	at org.openhab.binding.shelly.internal.coap.ShellyCoapServer$ShellyStatusListener.handleRequest(ShellyCoapServer.java:74) ~[?:?]
	at org.eclipse.californium.core.server.ServerMessageDeliverer.deliverRequest(ServerMessageDeliverer.java:108) ~[bundleFile:?]
	at org.eclipse.californium.core.network.stack.BaseCoapStack$StackTopAdapter.receiveRequest(BaseCoapStack.java:200) ~[bundleFile:?]
	at org.eclipse.californium.core.network.stack.AbstractLayer.receiveRequest(AbstractLayer.java:84) ~[bundleFile:?]
	at org.eclipse.californium.core.network.stack.AbstractLayer.receiveRequest(AbstractLayer.java:84) ~[bundleFile:?]
	at org.eclipse.californium.core.network.stack.BlockwiseLayer.receiveRequest(BlockwiseLayer.java:398) ~[bundleFile:?]
	at org.eclipse.californium.core.network.stack.ReliabilityLayer.receiveRequest(ReliabilityLayer.java:272) ~[bundleFile:?]
	at org.eclipse.californium.core.network.stack.AbstractLayer.receiveRequest(AbstractLayer.java:84) ~[bundleFile:?]
	at org.eclipse.californium.core.network.stack.BaseCoapStack.receiveRequest(BaseCoapStack.java:133) ~[bundleFile:?]
	at org.eclipse.californium.core.network.CoapEndpoint$1.receiveRequest(CoapEndpoint.java:267) ~[bundleFile:?]
	at org.eclipse.californium.core.network.UdpMatcher$2.run(UdpMatcher.java:256) [bundleFile:?]
	at org.eclipse.californium.elements.util.SerialExecutor$1.run(SerialExecutor.java:276) [bundleFile:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by markus7017** [comment]
It seems that the color updates doesn't work in general, even for other colors than white.
The reason: A chanell with item type Dimmer requires a PercentType, not the int value from the Coap update value.

**Comment by markus7017** [comment]
will be included in PR #6764

**Comment by markus7017** [comment]
fixed & verified

**Comment by markus7017** [comment]
will be included in PR #6985

**Comment by markus7017** [comment]
included in version 2.5.4
